flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/cli/CliClient.java

```diff
@@ -299,15 +307,23 @@ private void callOperation(Operation operation) {
         } else if (operation instanceof HelpOperation) {
             // HELP
             callHelp();
+        } else if (operation instanceof BeginStatementSetOperation) {
+            // BEGIN STATEMENT SET
+            callBeginStatementSet();
+        } else if (operation instanceof EndStatementSetOperation) {
+            // END
+            callEndStatementSet();
+        } else if (operation instanceof CatalogSinkModifyOperation) {
+            // INSERT INTO/OVERWRITE
+            callInsert((CatalogSinkModifyOperation) operation);
+        } else if (isStatementSetMode) {
```

Comment:
This looks really hack and hard to maintain what statement are not allowed in statement set. I suggest to check statements at the beginning of this method. 

```java
      if (isStatementSetMode) {
            // check the current operation is allowed in STATEMENT SET
            if (!(operation instanceof CatalogSinkModifyOperation
                    || operation instanceof EndStatementSetOperation)) {
                printError(MESSAGE_STATEMENT_SET_SQL_EXECUTION_ERROR);
                return;
            }
        }
```